### PR TITLE
Fix intermittent test failure related to file blob removal

### PR DESCRIPTION
**Pull request by mikebaumann**
The indexer test test_indexed_file_access_error formerly removed
the a file blob associated with a test bundle to verify the indexer
handles this case as expected. However, the file blob was for a commonly used
test file, and intermittent test failures were occurring when run using a shared test bucket.
The test was changed to instead create a bundle manifest entry such that the file blob
reference is invalid. The code and test semantics are the same.

### CODE REVIEW

File: tests/test_indexer.py [hunk 1]
```diff
@@ -327,6 +347,65 @@ def get_search_results(cls, query, expected_hit_count):
...
+    def get_bundle_id(self):
+        return f'{self.bundle_id}.{self.bundle_version}'
+
+    def add_file(self, bucket_name, name, indexed, file_id):
+        # Add the existing file to the bundle manifest
+        s3 = boto3.resource("s3")
+        file_manifest_string = s3.Object(bucket_name=bucket_name,
+                                         key=f'files/{file_id}').get()['Body'].read().decode('utf-8')
+        file_manifest = json.loads(file_manifest_string, encoding="utf-8")
+        file_uuid, _, file_version = file_id.partition(".")
```
Comment (ttung) [inline tests/test_indexer.py]: `file_uuid, file_version = file_id.split(".")`

File: tests/test_indexer.py [hunk 2]
```diff
@@ -152,9 +155,9 @@ def test_indexed_file_unparsable(self):
                                                          files=smartseq2_paried_ends_indexed_file_list)
 
     def test_indexed_file_access_error(self):
-        bundle_key = self.load_test_data_bundle_for_path('fixtures/smartseq2/paired_ends')
-        filename = "project.json"
-        deleteFileBlob(bundle_key, filename)
+        filename = "inaccessible_file.json"
```
Comment (ttung) [inline tests/test_indexer.py]: rename this var to inaccessible_file_name or something like that.

File: tests/test_indexer.py [hunk 3]
```diff
@@ -327,6 +347,65 @@ def get_search_results(cls, query, expected_hit_count):
                 time.sleep(0.5)
 
 
+class BundleBuilder:
```
Comment (ttung) [inline tests/test_indexer.py]: docblock

File: tests/test_indexer.py (hunk 1 above)
Comment (mikebaumann) [inline tests/test_indexer.py]: This doesn't seem to work due to two variables on the left side and three substrings from splitting on '.'
```
file_uuid, file_version = '6011caee-ebde-4b2b-82f1-bb75d303fa41.2017-08-07T150159.427496Z'.split(".")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```
Although it works if `maxsplit=1` is specified:
```
file_uuid, file_version = '6011caee-ebde-4b2b-82f1-bb75d303fa41.2017-08-07T150159.427496Z'.split(".", maxsplit=1)
```
@kislyuk used partition in https://github.com/HumanCellAtlas/data-store/pull/291 so I used it here as well.

File: tests/test_indexer.py (hunk 2 above)
Comment (mikebaumann) [inline tests/test_indexer.py]: Yes, that is a better name. Done.

File: tests/test_indexer.py (hunk 3 above)
Comment (mikebaumann) [inline tests/test_indexer.py]: Added a class docstring. It is brief, given this is a test class used within a single test file.

**Comment by mikebaumann** [comment]
As requested in the summary comment, I also changed the use of quotes as string delimiters to be consistent. This test file now uses single quote consistently as the string delimiter. Please let me know if we have more specific conventions regarding the use of single or double quote as string delimiters.

File: tests/test_indexer.py (hunk 1 above)
Comment (ttung) [inline tests/test_indexer.py]: Ahh. `file_uuid, file_version = file_id.split(".", 1)` should work, but I'll leave it to your discretion.

**Comment by ttung** [comment]
> As requested in the summary comment, I also changed the use of quotes as string delimiters to be consistent. This test file now uses single quote consistently as the string delimiter. Please let me know if we have more specific conventions regarding the use of single or double quote as string delimiters.

I don't recall where we landed on what used single or double quotes, but I believe it was dictionary string keys = single quotes, other strings = double quotes.

The safe way to make this change is to do a predecessor diff that fixes the style of the existing code, and then your changes stacked on top of that.

File: tests/test_indexer.py [hunk 4]
```diff
@@ -327,6 +347,72 @@ def get_search_results(cls, query, expected_hit_count):
...
+    """
+    Utility class for creating a new bundle for testing purposes.
```
Comment (ttung) [inline tests/test_indexer.py]: `Creating a new bundle manifest`

File: tests/test_indexer.py (hunk 3 above)
Comment (ttung) [inline tests/test_indexer.py]: IMO, we should consider making this a general purpose class.  bundle.py would probably benefit from it.

**Comment by kislyuk** [comment]
Closing because this was merged as #329. Please reopen if you need this for any reason.

**Pull request closed without merging**